boards/riscv/qemu_riscv32/Kconfig.board

```diff
@@ -5,15 +5,18 @@ config BOARD_QEMU_RISCV32
 	depends on SOC_RISCV_VIRT
 	select QEMU_TARGET
 	select CPU_HAS_FPU
+	select RISCV_ISA_RV32I
 
 config BOARD_QEMU_RISCV32_SMP
 	bool "QEMU RISCV32 SMP target"
 	depends on SOC_RISCV_VIRT
 	select QEMU_TARGET
 	select CPU_HAS_FPU
+	select RISCV_ISA_RV32I
```

Comment:
The RISC-V SoC layer seems to be a mess. There should probably be separate SoC symbols for the 32-bit and 64-bit variants: e.g. `SOC_SERIES_RISCV_VIRT_32` and `SOC_SERIES_RISCV_VIRT_64`, which select `RISCV_ISA_RV32I` and `RISCV_ISA_RV64I`, respectively.

Also, I have no idea why this `depends on SOC_RISCV_VIRT` instead of selecting it. The sane way to do it would be: board selects SoC, SoC selects arch core, arch core selects arch family and so on.

Anyways, that is beyond the scope of this PR and something for later ...

Reply:
yeah, this is indeed the only board config that I had to touch for exactly that reason. Not optimal.